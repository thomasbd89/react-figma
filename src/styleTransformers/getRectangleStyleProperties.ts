import { GeometryStyleProperties } from './transformGeometryStyleProperties';
import { YogaStyleProperties } from '../yoga/YogaStyleProperties';
import { LayoutStyleProperties } from './transformLayoutStyleProperties';
import { BlendStyleProperties } from './transformBlendProperties';
import { BorderStyleProperties } from './transformBorderProperties';

const rectangleStyleProperties = [
    'backgroundColor',
    'backgroundImage',
    'backgroundSize',
    'blendMode',
    'shadowColor',
    'shadowOffset',
    'shadowOpacity',
    'shadowRadius',
    'borderColor',
    'borderWidth',
    'borderRadius',
    'borderBottomLeftRadius',
    'borderBottomRightRadius',
    'borderTopLeftRadius',
    'borderTopRightRadius'
];

export const getRectangleStyleProperties = (
    style: Partial<
        GeometryStyleProperties &
            YogaStyleProperties &
            BorderStyleProperties &
            LayoutStyleProperties &
            BlendStyleProperties
    >
): Partial<GeometryStyleProperties & BlendStyleProperties & BorderStyleProperties> => {
    const result = {};
    rectangleStyleProperties.forEach(prop => {
        if (style[prop]) {
            result[prop] = style[prop];
        }
    });
    return result;
};
