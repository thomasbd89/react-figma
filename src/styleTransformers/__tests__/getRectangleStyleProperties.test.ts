import { getRectangleStyleProperties } from '../getRectangleStyleProperties';
import { StyleSheet } from '../..';

describe('getRectangleStyleProperties', () => {
    it('getRectangleStyleProperties', () => {
        const styles = StyleSheet.create({
            view: {
                width: '100%',
                height: 50,
                opacity: 0.9,
                borderRadius: 25,
                backgroundColor: '#000000',
                shadowColor: 'rgba(217, 217, 217, 0.5)',
                shadowOffset: {
                    width: 0,
                    height: 5
                },
                shadowRadius: 10,
                shadowOpacity: 1,
                alignItems: 'center',
                flexDirection: 'row',
                paddingLeft: 8,
                paddingRight: 8
            }
        });
        const result = getRectangleStyleProperties(styles.view);
        expect(result).toMatchSnapshot();
    });
});
