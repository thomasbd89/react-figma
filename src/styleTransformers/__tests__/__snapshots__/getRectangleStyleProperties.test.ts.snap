// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRectangleStyleProperties getRectangleStyleProperties 1`] = `
Object {
  "backgroundColor": "#000000",
  "borderRadius": 25,
  "shadowColor": "rgba(217, 217, 217, 0.5)",
  "shadowOffset": Object {
    "height": 5,
    "width": 0,
  },
  "shadowOpacity": 1,
  "shadowRadius": 10,
}
`;
